flink-core/src/main/java/org/apache/flink/core/fs/AutoCloseableRegistry.java

```diff
@@ -50,8 +50,15 @@ public class AutoCloseableRegistry
 
     private static final Object DUMMY = new Object();
 
+    private final boolean reverse;
+
     public AutoCloseableRegistry() {
+        this(true);
+    }
+
+    public AutoCloseableRegistry(boolean reverse) {
         super(new LinkedHashMap<>());
+        this.reverse = reverse;
```

Comment:
nit: it would be helpful to have javadoc to define what we mean by the reverse parameter.